testing/internal/e2e/tests/static/credential_store_test.go

```diff
@@ -67,13 +79,29 @@ func TestCliStaticCredentialStore(t *testing.T) {
 	err = json.Unmarshal(output.Stdout, &newSessionAuthorizationResult)
 	require.NoError(t, err)
 
-	newSessionAuthorization := newSessionAuthorizationResult.Item
-	retrievedUser, ok := newSessionAuthorization.Credentials[0].Credential["username"].(string)
-	require.True(t, ok)
-	retrievedPassword, ok := newSessionAuthorization.Credentials[0].Credential["password"].(string)
-	require.True(t, ok)
-	assert.Equal(t, c.TargetSshUser, retrievedUser)
-	assert.Equal(t, "password", retrievedPassword)
+	brokeredCredentials := make([]map[string]any, 0, 3)
+	for _, credential := range newSessionAuthorizationResult.Item.Credentials {
+		brokeredCredentials = append(brokeredCredentials, credential.Credential)
+	}
+
+	// Prepare expected credentials
+	testCredentialsJson, err := os.ReadFile(testCredentialsFile)
+	require.NoError(t, err)
+	var expectedJsonCredentials map[string]any
+	err = json.Unmarshal(testCredentialsJson, &expectedJsonCredentials)
+	require.NoError(t, err)
+
+	sshPrivateKeyFileContent, err := os.ReadFile(c.TargetSshKeyPath)
+	require.NoError(t, err)
+	sshPrivateKey := strings.TrimSuffix(string(sshPrivateKeyFileContent), "\n")
+
+	expectedCredentials := []map[string]any{
+		{"username": c.TargetSshUser, "password": testPassword},
+		{"username": c.TargetSshUser, "private_key": sshPrivateKey},
+		expectedJsonCredentials,
+	}
+
+	assert.ElementsMatch(t, expectedCredentials, brokeredCredentials)
```

Comment:
One thing I just thought about. If this assertion fails, this would print out the contents of the private key, would I don't think we want to do. Should we do something like generate a new key here as opposed to using the key in `c.TargetSshKeyPath`?

Reply:
That makes perfect sense. I've added a test .pem file generation and clean up for every test execution